File: src/main/java/com/wikia/tibia/repositories/SpellRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.SpellGateway;
import com.wikia.tibia.objects.Spell;

public class SpellRepository extends WikiObjectRepository {

    public SpellRepository() {
        super(Spell.class, new SpellGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/SpellRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Spell

class SpellRepository : WikiObjectRepository<Spell>(
    wikiObjectClass = Spell::class.java,
    wikiObjectGateway = WikiObjectGateway(Contract.SPELLS)
)
